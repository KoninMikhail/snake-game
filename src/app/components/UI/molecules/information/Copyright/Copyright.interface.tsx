// Copyright
// ----------------------------------------------------------------------

export interface ICopyrightProps {
    author: string;
    creationYear: number;
    license: string;
}
