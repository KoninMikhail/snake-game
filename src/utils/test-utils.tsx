import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
