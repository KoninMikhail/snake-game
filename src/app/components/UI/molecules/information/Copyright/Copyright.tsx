import { Text } from '@ui/atoms/typography/body/Text/Text';
import { StyledCopyright } from './style/StyledCopyright';
import { ICopyrightProps } from './Copyright.interface';
import { getYearRange } from './utils/Copyright.utils';
import { useMemo } from 'react';

/**
 * @name Copyright
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for creation copyright text.
 *
 * @param author
 * @param creationYear - If the year of creation is the same as the current year component
 * will return single number in a text else {creation-year} - {current year}
 * @param licence
 * @returns {JSX.Element}
 */
export const Copyright = ({ author, creationYear, license, ...rest }: ICopyrightProps) => {
    const projectYears = useMemo(() => getYearRange(creationYear), [creationYear]);
    return (
        <StyledCopyright {...rest}>
            <Text>{`Â© ${projectYears}  |  ${author}. ${license}.`}</Text>
        </StyledCopyright>
    );
};
