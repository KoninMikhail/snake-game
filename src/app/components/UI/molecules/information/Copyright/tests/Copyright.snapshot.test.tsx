import {Copyright} from '@ui/molecules/information/Copyright/Copyright';
import {renderWithProviders} from '@utils/test-utils';

/**
 * Test
 */
describe('Copyright', () => {
    const year = new Date('July 20, 69 00:20:18').getFullYear();
    const license = 'GNUv2';
    const author = 'Mickey Mouse';

    it('equal snapshot', () => {
        const { container } = renderWithProviders(
            <Copyright author={author} creationYear={year} license={license} />
        );
        expect(container).toMatchSnapshot();
    });
});
