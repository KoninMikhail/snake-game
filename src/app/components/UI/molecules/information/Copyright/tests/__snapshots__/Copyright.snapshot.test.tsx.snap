// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright equal snapshot 1`] = `
.c1 {
  color: white;
  font-size: 14px;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
}

.c0 {
  display: block;
}

@media (min-width:768px) {
  .c1 {
    font-size: 16px;
  }
}

@media (min-width:1200px) {
  .c1 {
    font-size: 1.1vw;
  }
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Â© 1969-2023  |  Mickey Mouse. GNUv2.
    </span>
  </div>
</div>
`;
